lib/msal-browser/src/interaction_client/SilentIframeClient.ts

```diff
@@ -44,6 +44,7 @@ export class SilentIframeClient extends StandardInteractionClient {
             ...request,
             prompt: PromptValue.NONE
         }, InteractionType.Silent);
+        this.browserStorage.updateCacheEntries(silentRequest.state, silentRequest.nonce, silentRequest.authority, silentRequest.loginHint || "", silentRequest.account || null);
```

Comment:
Can we put this in `initializeAuthorizationRequest`?

Can you also explain what this is for?

Reply:
This is to update temporary cache values. Using InitializeAuthorizationRequest makes it harder to control when the cache values are updated. For some broker scenarios, we don't want to update cache entries when we initialize the request, but after we send the request up to the broker.

Reply:
I see now that's where it used to be... Can we add a comment explaining this? This feels like something that may get moved back at some point without realizing what it affects. Not sure I like taking a dependency on implementation details like this

Reply:
I have the same inclination that this may get "optimized" at some point. 

Least case add a comment, Best case make a commented helper API with instructions regarding the sequence , 
`processReqBeforeDispatch(browserStorage, request: BaseReqType)`.

May prevent someone moving this later that way and any other functionality we may add before request dispatch for broker. Could be an overkill, will let you do the trade off.